### PR TITLE
feat(Overview): onlyOverview - External Compositions and API Reference

### LINKED ISSUE #7332: Properties created in the component are shown as "signature"
### Describe the bug

When I created and exported the component with its properties. Property shown as "signature" instead of shown correct type.

### Steps to Reproduce

1. Created a component and it's properties.

image

2. Check the table created by bit in Overview Section.

image

3. The type created as signature

### Expected Behavior

The property type must be shown.

image

### Specifications

- Bit version: 0.0.992
- Workspace type: harmony
- Node version: 16

### Additional context

This problem ocurred with functions and objects.

### PR DESCRIPTION
**Pull request by luvkapur**
This PR introduces a new `onlyOverview` preview option, enhancing the Overview section with a dedicated API Reference subsection and `useNameParam` preview option.

Key Features:

`onlyOverview` Preview Option:

Enabled via the preview data of a component's environment, this feature is automatically activated for all components tagged with this Bit version onwards.
It refines the Overview screen to render only the main documentation and examples within the iframe, while client-side mounting the Compositions Carousel along with the new API Reference subsection.

API Reference Subsection:

This update replaces the previous properties table, which was limited to displaying props for React components only.
The new API Reference table leverages our internal Schema Extractor (also powering the API Reference tab) to highlight the relevant APIs for your component, utilizing the @exports jsdoc tag.
By default, it presents an API Table of Contents, collating all exported APIs from the component's main file and linking them to their detailed information in the API Reference Tab.

`useNameParam` Preview Option: 

Enabled via the preview data of a component's environment, this feature is automatically activated for all components tagged with this Bit version onwards.
It sets the selected composition via name url param to view selected compositions, previously this was set with a combination of two preview options (`includesEnvTemplate` and `doesScaling`)

Compositions Order Bug Fix:

This PR also fixes the bug where the compositions were incorrectly ordered. Now all compositions are ordered according to the order they are defined. 

Loading Overview Docs Iframe:

This PR also adds a skeleton loading spinner while the docs iframe loads

fixes #7332

### CODE REVIEW

**Review by GiladShoham — CHANGES_REQUESTED** [review]

**Review by davidfirst — APPROVED** [review]

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 1]
```diff
@@ -436,6 +495,42 @@ export class PreviewMain {
     return !!previewData?.skipIncludes;
   }
 
+  /**
+   * check if the component preview should only include the overview (skipping rendering of the compostions and properties table)
+   */
+  async includesOnlyOverview(component: Component): Promise<boolean> {
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:501]: rename to get*

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 2]
```diff
@@ -436,6 +495,42 @@ export class PreviewMain {
...
+      return envSupportOnlyOverview;
+    }
+    const previewData = this.getPreviewData(component);
+    return previewData?.onlyOverview ?? false;
+  }
+
+  /**
+   * check if the component preview should include the name query param
+   */
+  async useNameParam(component: Component): Promise<boolean> {
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:520]: rename to get*

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 3]
```diff
@@ -436,6 +495,42 @@ export class PreviewMain {
...
+    const inWorkspace = await this.workspace?.hasId(component.id);
+    if (inWorkspace) {
+      if (this.envs.isUsingCoreEnv(component)) {
+        return true;
+      }
+      const envComponent = await this.envs.getEnvComponent(component);
+      const envSupportOnlyOverview = await this.doesEnvIncludesOnlyOverview(envComponent);
+      return envSupportOnlyOverview;
+    }
+    const previewData = this.getPreviewData(component);
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:513]: read from the data only

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 4]
```diff
@@ -436,6 +495,42 @@ export class PreviewMain {
...
+    const inWorkspace = await this.workspace?.hasId(component.id);
+    if (inWorkspace) {
+      if (this.envs.isUsingCoreEnv(component)) {
+        return true;
+      }
+      const envComponent = await this.envs.getEnvComponent(component);
+      const envSupportNameParam = await this.doesEnvUseNameParam(envComponent);
+      return envSupportNameParam;
+    }
+    const previewData = this.getPreviewData(component);
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:530]: read from the data only

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 5]
```diff
@@ -296,15 +351,17 @@ export class PreviewMain {
...
+
     const previewAspectConfig = this.getPreviewAspectConfig(envComponent);
 
     const data = {
       // default to true if the env doesn't have a preview config
       isScaling: previewAspectConfig?.isScaling ?? true,
-      // disalbe it for now, we will re-enable it later
-      // skipIncludes: true,
+      onlyOverview: true,
+      useNameParam: true,
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:364]: change to support*

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 6]
```diff
@@ -243,6 +257,42 @@ export class PreviewMain {
...
+  }
+
+  /**
+   * check if the current version of env component supports name query param
+   * @param envComponent
+   * @returns
+   */
+  doesEnvUseNameParam(envComponent: Component): boolean {
+    const previewData = this.getPreviewData(envComponent);
+    return !!previewData?.useNameParam;
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:277]: change to support*

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 7]
```diff
@@ -243,6 +257,42 @@ export class PreviewMain {
...
   }
 
+  /**
+   * check if the current version of env component supports skipping other included previews
+   * @param envComponent
+   * @returns
+   */
+  doesEnvIncludesOnlyOverview(envComponent: Component): boolean {
+    const previewData = this.getPreviewData(envComponent);
+    return !!previewData?.onlyOverview;
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:267]: change to support*

File: scopes/pipelines/builder/builder.main.runtime.ts [hunk 8]
```diff
@@ -148,9 +148,64 @@ export class BuilderMain {
...
+        ];
+      })
+    );
+
+    const harmonyCompIdsWithEnvIdMap = new Map(harmonyCompIdsWithEnvId);
+
+    const compsToDeleteOnlyOverviewPreviewData = harmonyComps.filter((comp) => {
+      const envData: { envId: string; inWs: boolean; lastTaggedEnvHasOnlyOverview: boolean } | undefined =
+        harmonyCompIdsWithEnvIdMap.get(comp.id.toString());
+      return envData?.inWs && !envData?.lastTaggedEnvHasOnlyOverview;
```
Comment (GiladShoham) [inline scopes/pipelines/builder/builder.main.runtime.ts:199]: what if the env is also tagged in the process?

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/preview/preview/preview.main.runtime.tsx [hunk 9]
```diff
@@ -436,6 +494,23 @@ export class PreviewMain {
     return !!previewData?.skipIncludes;
   }
 
+  /**
+   * check if the component preview should only include the overview (skipping rendering of the compostions and properties table)
+   */
+  async getOnlyOverview(component: Component): Promise<boolean> {
+    // if (!this.config.onlyOverview) return false;
```
Comment (GiladShoham) [inline scopes/preview/preview/preview.main.runtime.tsx:501]: Is this commented by purpose?

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**